src/index/metadata.rs

```diff
@@ -203,6 +205,32 @@ fn read_flac(path: &Path) -> Result<SongTags> {
 	})
 }
 
+#[cfg_attr(feature = "profile-index", flame)]
+fn read_mp4(path: &Path) -> Result<SongTags> {
+	let tag = mp4ameta::Tag::read_from_path(path)?;
+	let mut tags = SongTags {
+		artist: None,
+		album_artist: None,
+		album: None,
+		title: None,
+		duration: None,
+		disc_number: None,
+		track_number: None,
+		year: None,
+	};
+	tags.artist = tag.artist().map(|v| v.to_string());
+	tags.album_artist = tag.album_artist().map(|v| v.to_string());
+	tags.album = tag.album().map(|v| v.to_string());
+	tags.title = tag.title().map(|v| v.to_string());
+	tags.duration = tag.duration().map(|v| v as u32);
+	tags.disc_number = tag.disk_number().0.and_then(|d| Some(d as u32));
+	tags.track_number = tag.track_number().0.and_then(|d| Some(d as u32));
+	let year = tag.year().and_then(|v| Some(v.parse::<i32>().ok()));
+	tags.year = match year { None => None, Some(year) => year};
```

Comment:
I think these two lines can be simplified into:

    tags.year = tag.year().and_then(|v| v.parse::<i32>().ok());

`.ok()` already returns an `Option` so the extra `Some()` isn't adding anything and requires you to use the `match` on the following line.

Reply:
Will do.

Reply:
Done.